src/test/java/com/nashtech/controller/ReactiveDataControllerTest.java

```diff
@@ -0,0 +1,122 @@
+package com.nashtech.controller;
+
+import com.nashtech.model.Car;
+import com.nashtech.model.CarBrand;
+import com.nashtech.service.ReactiveDataService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+public class ReactiveDataControllerTest {
+
+    @Mock
+    private ReactiveDataService reactiveDataService;
+
+    @InjectMocks
+    private ReactiveDataController reactiveDataController;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void testGetAllBrands_ReactiveDataServiceReturnsError() {
+        // Setup
+        when(reactiveDataService.getAllBrands()).thenReturn(Flux.error(new Exception("message")));
+
+        // Run the test
+        final Flux<CarBrand> result = reactiveDataController.getAllBrands();
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectErrorMatches(throwable -> throwable instanceof Exception && throwable.getMessage().equals("message"))
+                .verify();
+    }
+
+    @Test
+    void testGetAllBrands_ReactiveDataServiceReturnsNoItem() {
+        // Setup
+        when(reactiveDataService.getAllBrands()).thenReturn(Flux.empty());
+
+        // Run the test
+        final Flux<CarBrand> result = reactiveDataController.getAllBrands();
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    void testGetCarsByBrand_ReactiveDataServiceReturnsError() {
+        // Setup
+        when(reactiveDataService.getCarsByBrand("brand")).thenReturn(Flux.error(new Exception("message")));
```

Comment:
Don't give "message" as the error message for the exception. Could you add a descriptive error message?